### PR TITLE
fix:grpc client request timeout processing

### PR DESCRIPTION
**Pull request by kzhan**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: protocol/grpc/client.go [hunk 1]
```diff
@@ -123,3 +134,13 @@ func getInvoker(impl interface{}, conn *grpc.ClientConn) interface{} {
 	res := method.Call(in)
 	return res[0].Interface()
 }
+
+//client timeout
+func UnaryClientTimeoutInterceptor(timeout time.Duration) grpc.UnaryClientInterceptor {
```
Comment (fangyincheng) [inline protocol/grpc/client.go:139]: Why does it start with capital？

**Comment by fangyincheng** [comment]
test passed？id timeout valid？

File: protocol/grpc/client.go (hunk 1 above)
Comment (zouyx) [inline protocol/grpc/client.go:139]: I think it is private method for `grpc` package, no need to export

**Comment by kzhan** [comment]
> test passed？id timeout valid？

The test passed and is online

**Comment by kzhan** [comment]
> test passed？id timeout valid？

The test passed and is online

File: protocol/grpc/client.go (hunk 1 above)
Comment (kzhan) [reply protocol/grpc/client.go:139]: I'm sorry, let me fix that

Comment (kzhan) [reply protocol/grpc/client.go:139]: > I think it is private method for `grpc` package, no need to export

I'm sorry, let me fix that

**Comment by zouyx** [comment]
> > test passed？id timeout valid？
> 
> The test passed and is online

You can check it offline.
On MacOS, open command line and run ```make verify```.
On Windows install sub-system , like ubuntu then open command line and run ```make verify```.

**Comment by kzhan** [comment]
> > > test passed？id timeout valid？
> > 
> > 
> > The test passed and is online
> 
> You can check it offline.
> On MacOS, open command line and run `make verify` in root dir.
> On Windows install sub-system like ubuntu, then open command line and run `make verify` in root dir.

ok

**Review by AlexStocks — APPROVED** [review]

**Pull request closed without merging**